provision.yml

```diff
@@ -7,147 +7,181 @@
     tags:
      - always
 
-
-- hosts: loadbalancer:loadbalancer-vm:php-apps:java-apps:storage:oidc:dbcluster:sysloghost:elk:lifecycle
+- hosts: base
   gather_facts: yes
   become: true
   roles:
-    - { role: common,       tags: ['common'] }
-    - { role: rsyslog,      tags: ['rsyslog'] }
-    - { role: iptables, when: iptables_enable | bool,  tags: ['iptables'] }
-
-- hosts: selfsigned_certs
-  gather_facts: no
-  become: true
-  roles:
-    - { role: selfsigned_certs, tags: ['selfsigned_certs'] } 
-
-- hosts: php-apps-vm
-  become: true
-  roles:
-    - { role: hosts,     tags: ['hosts'] }
-    - { role: httpd,     tags: ['httpd'] }
-    - { role: welcome,   tags: ['welcome'] }
-
-- hosts: loadbalancer-vm
-  gather_facts: true
-  become: true
-  roles:
-    - { role: haproxy, tags: ['lb'] }
+    - role: common
+      tags: ['core', 'base', 'common']
+    - role: rsyslog
+      tags: ['core', 'base', 'rsyslog']
+    - role: iptables
+      when:
+        - iptables_enable | bool
+      tags: ['core', 'base', 'iptables']
+    - role: selfsigned_certs
+      when:
+        - use_selfsigned_certs
+      tags: ['core', 'base', 'selfsigned_certs']
 
-- hosts: loadbalancer_ha
-  gather_facts: true
+- hosts: frontend
   become: true
   roles:
-    - { role: keepalived, tags: ['keepalived'] }
-    - { role: bind, tags: ['bind'] }
+    - role: hosts
+      tags: ['core', 'frontend', 'hosts']
+    - role: httpd
+      tags: ['core', 'frontend', 'httpd']
+    - role: welcome
+      tags: ['core', 'frontend', 'welcome']
```

Comment:
This role is optional as well. It would be beneficial to get this role behind a feature toggle as well

Reply:
Done!